charms/kfp-api/src/charm.py

```diff
@@ -588,7 +588,17 @@ def _on_remove(self, _):
 
     def _on_update_status(self, _):
         """Update status actions."""
-        self._on_event(_)
+        try:
+            self._on_event(_)
+        except ErrorWithStatus as err:
+            raise err
```

Comment:
do we really raise ?

this will put the unit into error state

Reply:
I think you are correct. Should we just return or pass in this case?

Reply:
Fixed.